File: backEnd/src/main/java/com/project/smfinance/entity/Customer.java
package com.project.smfinance.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Lob;
import jakarta.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Customer extends BaseEntity {

  @Column(nullable = false)
  private String name;

  @Column(nullable = false)
  private String phoneNumber;

  @Column(nullable = true)
  private String altPhoneNumber;

  @Column(nullable = false, columnDefinition = "LONGTEXT")
  @Lob
  private String address;

  @Column(nullable = true, unique = true)
  private String email;

  @Column(nullable = false)
  private String occupation;

  @OneToMany(
      mappedBy = "customer",
      cascade = CascadeType.ALL,
      orphanRemoval = true,
      fetch = FetchType.EAGER)
  private List<CustomerDocument> documents = new ArrayList<>();
}

File: frontend/src/types/customer.ts
export interface CustomerDocumentRequest {
  documentType: string;
  documentPath: string;
  documentKey: string;
  documentContentType: string;
}

export interface CustomerRequest {
  name: string;
  phoneNumber: string;
  altPhoneNumber: string,
  address: string;
  email: string;
  occupation: string;
  documents: CustomerDocumentRequest[];
}

export interface CustomerDocumentResponse extends CustomerDocumentRequest {
  id: number;
}

export interface CustomerResponse extends CustomerRequest {
  id: number;
  documents: CustomerDocumentResponse[];
}
